File: app/src/main/java/com/dingyi/MyLuaApp/base/BaseFragment.java
package com.dingyi.MyLuaApp.base;

import android.content.res.Configuration;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.viewbinding.ViewBinding;

import com.dingyi.MyLuaApp.utils.TextUtils;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

public abstract class BaseFragment<T extends ViewBinding> extends Fragment {
    protected Handler mHandler = new Handler(Looper.getMainLooper());

    private T mBinding;

    private BaseActivity<?> mActivity;

    private Event mEvents;


    public BaseFragment(BaseActivity<?> activity) {
        mActivity = activity;
    }

    public void setEvent(Event event) {
        this.mEvents = event;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        initView(getViewBinding(inflater, container));
        if (mEvents != null) {
            mEvents.onCreateView();
        }
        return getViewBinding(inflater, container).getRoot();
    }

    protected Handler getHandler() {
        return mHandler;
    }


    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);

    }

    protected abstract Class<T> getViewBindingClass();

    public T getViewBinding(LayoutInflater inflater, ViewGroup root) {
        if (mBinding == null) {
            try {
                Method method = getViewBindingClass().getDeclaredMethod("inflate", LayoutInflater.class, ViewGroup.class, Boolean.TYPE);
                method.setAccessible(true);
                mBinding = (T) method.invoke(null, inflater, root, false);
            } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
                TextUtils.printError(e.toString());
            }
        }
        return mBinding;
    }

    public T getViewBinding() {
        return mBinding;
    }

    protected abstract void initView(T binding);


    public interface Event {
        void onCreateView();
    }

}

File: app/src/main/kotlin/com/dingyi/myluaapp/base/BaseFragment.kt
package com.dingyi.myluaapp.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.LinearLayoutCompat
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.viewbinding.ViewBinding

/**
 * @author: dingyi
 * @date: 2021/8/4 21:55
 * @description:
 **/
abstract class BaseFragment<V : ViewBinding, T : ViewModel> : Fragment() {


    protected lateinit var viewBinding: V

    protected lateinit var viewModel: T


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)


        viewModel = ViewModelProvider(requireActivity(),ViewModelProvider.NewInstanceFactory.instance)[getViewModelClass()]


    }


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return getViewBindingImp(inflater, container).root.apply {
            layoutParams =
                LinearLayoutCompat.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        }
    }


    private fun getViewBindingImp(inflater: LayoutInflater, container: ViewGroup?): V {
        if (!this::viewBinding.isInitialized) {
            viewBinding = getViewBindingInstance(inflater, container)
        }
        return viewBinding
    }


    abstract fun getViewModelClass(): Class<T>

    abstract fun getViewBindingInstance(inflater: LayoutInflater, container: ViewGroup?): V




}